src/main/java/com/example/petbuddybackend/service/user/UserService.java

```diff
@@ -90,29 +87,39 @@ public ProfileData getProfileData(String email) {
     @Transactional
     public PhotoLinkDTO uploadProfilePicture(String username, MultipartFile profilePicture) {
         AppUser user = getAppUser(username);
-        Optional<PhotoLink> oldPhoto = photoService.findByNullableId(user.getProfilePictureBlob());
+        PhotoLink oldPhoto = user.getProfilePicture();
 
-        oldPhoto.ifPresent(photoService::deletePhoto);
+        if(oldPhoto != null) {
+            photoService.deletePhoto(oldPhoto);
+        }
 
         PhotoLink newPhoto = photoService.uploadPhoto(profilePicture);
-        user.setProfilePictureBlob(newPhoto.getBlob());
-        userRepository.save(user);
 
+        user.setProfilePicture(newPhoto);
+        userRepository.save(user);
         return photoMapper.mapToPhotoLinkDTO(newPhoto);
     }
 
     @Transactional
     public void deleteProfilePicture(String username) {
         AppUser user = getAppUser(username);
-        Optional<PhotoLink> profilePicture = photoService.findByNullableId(user.getProfilePictureBlob());
+        PhotoLink profilePicture = user.getProfilePicture();
```

Comment:
Pobranie PhotoLinka z usera do metody prywanej

Reply:
Nie wiem czy dobrze rozumiem. To jest do przeniesienia?
```java
        AppUser user = getAppUser(username);
        PhotoLink profilePicture = user.getProfilePicture();
``` 

Jak tak, to bym zostawił bo niżej potrzebuje usera:
```java
        user.setProfilePicture(null);
        userRepository.save(user);
```

Reply:
aha to sory, jest git